doc/en/user/source/styling/mbstyle/cookbook/lines.rst

```diff
@@ -0,0 +1,295 @@
+     - local-road
+   * - line.19
+     - Hill Street
+     - local-road
+   * - line.20
+     - Country Road
+     - local-road
+   * - line.21
+     - Main Street
+     - local-road
+   * - line.22
+     - Jani Lane
+     - local-road
+   * - line.23
+     - Shinbone Alley
+     - local-road
+   * - line.24
+     - State Street
+     - local-road
+   * - line.25
+     - River Road
+     - local-road
+
+:download:`Download the lines shapefile <artifacts/ysld_cookbook_line.zip>`
+
+.. _mbstyle_cookbook_lines_simpleline:
+
+Simple line
+-----------
+
+This example specifies lines be colored black with a thickness of 3 pixels.
+
+.. figure:: ../../sld/cookbook/images/line_simpleline.png
+
+   Simple line
+
+Code
+~~~~
+
+:download:`Download the "Simple line" MBStyle <artifacts/mbstyle_line_simpleline.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-line",
+    "layers":[
+      {
+        "id": "simple-line",
+        "type": "line",
+        "paint": {
+          "line-color": "#000000",
+          "line-width": "3"
+            }
+          }
+        ]
+    }
+
+Details
+~~~~~~~
+
+There is one layer style for this MBStyle, which is the simplest possible situation. Styling
+lines is accomplished in the paint (**lines 8-10**). **Line 9** specifies the color of the line to be
+black (``'#000000'``), while **line 10** specifies the width of the lines to be 3 pixels.
+
+
+Line with border
+----------------
+
+This example shows how to draw lines with borders (sometimes called "cased lines").
+In this case the lines are drawn with a 3 pixel blue center and a 1 pixel wide gray border.
+
+.. figure:: ../../sld/cookbook/images/line_linewithborder.png
+
+   Line with border
+
+Code
+~~~~
+
+:download:`Download the "Line with border" MBStyle <artifacts/mbstyle_line_borderedline.json>`
+
+.. code-block:: json
+  :linenos:
+
+  {
+    "version": 8,
+    "name": "simple-borderedline",
+    "layers":[
+      {
+            "id": "simple-borderedline",
+            "type": "line",
+            "layout": {
+              "line-cap": "round"
+            },
+            "paint": {
+              "line-color": "#333333",
+              "line-width": "5"
+            }
+          },
+          {
+            "id": "simple-line",
+            "type": "line",
+            "layout": {
+              "line-cap": "round"
+            },
+            "paint": {
+              "line-color": "#6699FF",
+              "line-width": "3"
+            }
+          }
+        ]
+    }
+
+
+Details
+~~~~~~~
+
+Lines in MBStyle have no notion of a "fill", only "stroke". Thus, unlike points or polygons, it is not possible to style the "edge" of the line geometry. It is, however, possible to achieve this effect by drawing each line twice: once with a certain width and again with a slightly smaller width. This gives the illusion of fill and stroke by obscuring the larger lines everywhere except along the edges of the smaller lines.

```

Comment:
This is kind of moot, since while the MBStyle fill and circle layers do allow for a line, they do not really allow any styling of that line (beyond whether the line is present or not).

(This section looks like it came from SLD/YSLD cookbook, where points / polygons do support styling lines)